### PR TITLE
New tasks

### PR DESCRIPTION
**Pull request by kolbytn**
Modifies distance task and adds location task.

### CODE REVIEW

File: Source/Holodeck/Tasks/Private/DistanceTask.cpp [hunk 1]
```diff
@@ -6,8 +6,15 @@
 void UDistanceTask::InitializeSensor() {
 	Super::InitializeSensor();
 
-	StartDistance = (GoalObject->GetActorLocation() - Parent->GetActorLocation()).Size();
-	NextDistance = StartDistance - Interval;
+	if (DistanceActor)
```
Comment (jaydenmilne) [inline Source/Holodeck/Tasks/Private/DistanceTask.cpp]: Could you put these in braces?

File: Source/Holodeck/Tasks/Private/DistanceTask.cpp [hunk 2]
```diff
@@ -41,38 +55,42 @@ void UDistanceTask::ParseSensorParms(FString ParmsJson) {
...
 // Called every frame
 void UDistanceTask::TickSensorComponent(float DeltaTime, ELevelTick TickType, FActorComponentTickFunction* ThisTickFunction) {
 	if (IsValid(Parent)) {
-		if (UseDistanceReward)
-			SetDistanceReward();
-		else
-			SetUnitReward();
+		float Distance = (DistanceLocation - this->GetComponentLocation()).Size();
+		if (DistanceActor)
+			Distance = (DistanceActor->GetActorLocation() - Parent->GetActorLocation()).Size();
```
Comment (jaydenmilne) [inline Source/Holodeck/Tasks/Private/DistanceTask.cpp:72]: Add braces here too

File: Source/Holodeck/Tasks/Private/LocationTask.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+					DistanceActor = *ActorItr;
+					break;
+				}
+			}
+		}
+
+		if (JsonParsed->HasTypedField<EJson::Array>("DistanceLocation")) {
+			TArray<TSharedPtr<FJsonValue>> LocationArray = JsonParsed->GetArrayField("DistanceLocation");
+			if (LocationArray.Num() == 3) {
+				double X;
```
Comment (jaydenmilne) [inline Source/Holodeck/Tasks/Private/LocationTask.cpp]: `double X, Y, Z;` ?

File: Source/Holodeck/Tasks/Private/LocationTask.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	Super::InitializeSensor();
+}
+
+// Allows sensor parameters to be set programmatically from client.
+void ULocationTask::ParseSensorParms(FString ParmsJson) {
+	Super::ParseSensorParms(ParmsJson);
+
+	TSharedPtr<FJsonObject> JsonParsed;
+	TSharedRef<TJsonReader<TCHAR>> JsonReader = TJsonReaderFactory<TCHAR>::Create(ParmsJson);
+	if (FJsonSerializer::Deserialize(JsonReader, JsonParsed)) {
```
Comment (jaydenmilne) [inline Source/Holodeck/Tasks/Private/LocationTask.cpp:16]: So if this doesn't work, it just silently fails?

**Review by jaydenmilne — CHANGES_REQUESTED** [review]

**Review by MaxRobinsonTheGreat — APPROVED** [review]
This looks soo good

**Review by jaydenmilne — APPROVED** [review]

**Pull request merged**